wrappers/src/shared_realm_cs.cpp

```diff
@@ -444,4 +448,45 @@ REALM_EXPORT SharedRealm* shared_realm_freeze(const SharedRealm& realm, NativeEx
     });
 }
 
+REALM_EXPORT Object* shared_realm_get_object_for_primary_key(SharedRealm& realm, TableKey tableKey, realm_value_t primitive, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() -> Object* {
+        realm->verify_thread();
+
+        auto table = realm->read_group().get_table(tableKey);
+        const std::string object_name(ObjectStore::object_type_for_table_name(table->get_name()));
+        auto& object_schema = *realm->schema().find(object_name);
```

Comment:
Add `find(TableKey)` in OS's Schema class and use that instead.

Reply:
I have just tried to implement this functionality as we discussed with @fealebenpae, by creating a map of `TableKey` to `ObjectSchema` when the `Schema` is created. Unfortunately the `table_key` field of `ObjectSchema` is not assigned yet in the constructor of `Schema`, and so we cannot create the map at that time. 
Given this, probably the fastest option would be to keep the code as it was as two operations are performed:

- `ObjectStore::object_type_for_table_name(table->get_name())` - O(1) time
- `realm->schema().find(object_name)` - O(log n) time (it uses binary search inside).

In alternative we can still implement find(TableKey) method, in two ways:

- We do a linear search - O(n) time
- We build the map the first time the method is called, and then use the map the following times - O(1) time + build time on first call

Reply:
The big-O estimates for the string case appear to ignore the string operations themselves. Not a C++ expert, but I don't believe the language uses any sort of pooling, which means that constructing the string itself and comparing it is a O(m) operation, which is exacerbated by the fact all table names start with `class_`.

Perhaps ping @fealebenpae for suggestions as that's ultimately code his team will review and maintain 😄

Reply:
As discussed with @nirinchev, we have decided to just use the linear search for now, then later we could eventually improve it